app/src/main/java/com/nevilleantony/prototype/downloadmanager/DownloadRepo.java

```diff
@@ -145,6 +145,10 @@ public void updateMinRange(String groupId, Long range, Long minRange) {
         stgApi.updateMinRange(groupId, range, minRange);
     }
 
+    public void insertAvailDownloads(String groupId, Long partNo){
+        stgApi.insertAvailDownload(groupId, partNo);
+    }
+
```

Comment:
1) This method should also update the `availablePartsMap`
2) There is already a method for this, see `addAvailablePart`, so is not needed